sdks/python/apache_beam/dataframe/frames.py

```diff
@@ -4073,6 +4077,10 @@ def fn_wrapper(x, *args, **kwargs):
             requires_partition_by=partitionings.Index(levels),
             preserves_partition_by=partitionings.Index(self._grouping_indexes)))
 
+  @frame_base.with_docs_from(DataFrameGroupBy)
+  def pipe(self, func, *args, **kwargs):
+    return func(self, *args, **kwargs)
```

Comment:
Note func can also be a tuple of `(callable, str)`, where str is the name of a kwarg to pass `self` to (see the [docs](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.pipe.html)). Could you implement and test that logic as well?

Reply:
Done 👍 , also added tests for these cases when used with a df, s, and gb.